posthog/api/query.py

```diff
@@ -79,6 +82,14 @@ def create(self, request, *args, **kwargs) -> Response:
                 data.query.model_dump(), data.filters_override.model_dump(), self.team
             )  # type: ignore
 
+        if data.variables_override is not None:
+            if isinstance(data.query, BaseModel):
```

Comment:
NIT: Is this check necessary for dashboard_filters too? Looks symmetric to the above logic but above logic doesn't check for mode_dump

Reply:
One of the endpoints this can come from parses the JSON into the model whereas the other doesn't - the filters override only comes from a single endpoint